src/components/ContentSidebar/ContentSidebar.js

```diff
@@ -769,38 +747,112 @@ class ContentSidebar extends PureComponent<Props, State> {
         });
     };
 
+    /**
+     * Sorts and formats assignments, and then adds them to their task. 
+     *
+
+     * @param {Task} task - Task to which the assignments belong
+     * @param {Task} assignments - List of task assignments
+     * @return {Task}
+     */
+    appendAssignmentsToTask(task: Task, assignments: Array<TaskAssignment>) {
+        if (!assignments) {
+            return task;
+        }
+
+        task.task_assignment_collection.entries = assignments.map((taskAssignment) => {
+            const { id, assigned_to, message, resolution_state } = taskAssignment;
+            return {
+                type: 'task_assignment',
+                id,
+                assigned_to,
+                message,
+                resolution_state
+            };
+        });
+
+        // Increment the assignment collection count by the number of new assignments
+        task.task_assignment_collection.total_count += assignments.length;
+        return task;
+    }
+
     /**
      * Adds a task to the tasks state and increases total_count.
      *
      * @param {Task} task - The newly created task from the API
+     * @param {SelectorItems} assignees - The list of users assigned to this task
+     * @param {Function} successCallback - Task create success callback
+     * @param {Function} errorCallback - Task create error callback
+
      * @return {void}
      */
-    createTaskSuccessCallback(task: Task): void {
-        const { tasks } = this.state;
-        if (tasks && tasks.entries) {
+    createTaskSuccessCallback(
```

Comment:
If there's any way you could abstract any of the internals of this function away, that would be great. It's starting to get pretty big

Reply:
moved out some error stuff